File: ClientApp/app/components/report/report.ts
import { IEntityWithIdName } from "../shared/shared-interfaces";

export interface IReportList {
    totalCount: number;
    reports: IReport[];
}

export interface IReport {
    reportGUID: string;
    name: string;
    query: IEntityWithIdName;
    creationDate: Date;
    author: IEntityWithIdName;
    modifyDate: Date;
    lastModifier: IEntityWithIdName;
    //style: string;
}

export interface IReportCreate {
    name: string;
    queryGUID: string;
    columns: string[];
    filter: string;
    rows: number;
    sort: IColumnSort;
    //Style: string;
}

export interface IColumnSort {
    columnName: string;
    direction: string;
}

//export class Report implements IReport {
//    constructor(public id: number,
//        public name: string,
//        public query: IEntityWithIdName,
//        public createdAt: string,
//        public createdBy: IEntityWithIdName,
//        public modifiedAt: string,
//        public modifiedBy: IEntityWithIdName,
//        public style: string) {
//    }

//    // műveletek
//}

File: ReportingWebApplication/Models/Entities/Report.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace szakdoga.Models
{
    public class Report : BaseEntity, IComparable
    {
        [Required]
        [StringLength(50)]
        public string ReportGUID { get; set; }

        public Query Query { get; set; }

        /// <summary>
        /// A grafikonok adataihoz
        /// </summary>
        public string Style { get; set; }

        /// <summary>
        /// :-vel tagolt string "tömb"
        /// </summary>
        public string Columns { get; set; }

        public string Filter { get; set; }
        public string Sort { get; set; }
        public int Rows { get; set; }
        public User LastModifier { get; set; }
        public User Author { get; set; }

        public int CompareTo(object obj)
        {
            Report report = obj as Report;
            if (report!=null)
            {
                return this.Name.CompareTo(report.Name);
            }
            return 0;
        }
    }
}